### ISSUE TITLE
False-positive object_name_linter for `%<-%`

### ISSUE THREAD

**Issue by bersbersbers** [jimhester/lintr#495]
This line

    `%<-%` <- zeallot::`%<-%`

raises

> Variable and function name style should be snake_case.

**Comment by russHyde** [comment]
@bersbersbers could you check if this is still an issue when using the current master. The object-name linter used by the `default_linters` has been modified to allow symbol-names.

**Comment by russHyde** [comment]
Cross-ref https://github.com/jimhester/lintr/pull/677

**Comment by bersbersbers** [comment]
Does not work:

```
> Rscript -e 'devtools::install_github("https://github.com/jimhester/lintr", force=T)'
Downloading GitHub repo jimhester/lintr@HEAD

✔  checking for file ‘/tmp/Rtmpuvgv6n/remotes56b02395a43a/jimhester-lintr-e759d2a/DESCRIPTION’ ...
─  preparing ‘lintr’:
✔  checking DESCRIPTION meta-information ...
─  installing the package to process help pages
─  checking for LF line-endings in source and make files and shell scripts (3.5s)
─  checking for empty or unneeded directories
─  building ‘lintr_2.0.1.9000.tar.gz’
   Warning: invalid uid value replaced by that for user 'nobody'
   
* installing *source* package ‘lintr’ ...
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (lintr)
> echo "\`%<-%\` <- zeallot::\`%<-%\`" > bug.R
> Rscript -e 'lintr::lint("bug.R")'
/home/berker/bug.R:1:1: style: Variable and function name style should be snake_case or symbols.
`%<-%` <- zeallot::`%<-%`
```

**Comment by AshesITR** [comment]
Thanks for trying. This should conform to the new `symbols` style, so that regex needs fixing 😄

**Comment by bersbersbers** [comment]
Works after
```
Rscript -e 'devtools::install_github("https://github.com/jimhester/lintr/pull/698")'
```

**Comment by MichaelChirico** [comment]
Great! Closing

**Comment by MichaelChirico** [comment]
Oh. I can't read.
